aioamqp/protocol.py

```diff
@@ -317,9 +321,13 @@ def channel(self, **kwargs):
         """Factory to create a new channel
 
         """
-        if not len(self.channels_ids):
-            raise exceptions.NoChannelAvailable()
-        channel_id = self.channels_ids.pop(0)
+        try:
+            channel_id = self.channels_ids_free.pop()
+        except KeyError:
+            if self.channels_ids_ceil >= amqp_constants.MAX_CHANNELS:
```

Comment:
The ids_ceil starts at zero, but MAX_CHANNELS is not a counter but an amount of available channel. I'm not sure if I' clear but I don't think we will get into this branch because the server won't give us any channel with ID "MAX_CHANNEL".

Its worth a test :p